src/Microsoft.Data/BaseDataFrameColumn.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    public abstract class BaseDataFrameColumn
+    {
+        public BaseDataFrameColumn(string name, long length = 0)
+        {
+            Length = length;
+            Name = name;
+        }
+        public long Length { get; protected set; }
+        public long NullCount { get; protected set; }
+        public string Name;
+
+        public virtual object this[long rowIndex] { get { throw new NotImplementedException(); } set { throw new NotImplementedException(); } }
+        public virtual object this[long startIndex, int length] { get { throw new NotImplementedException(); } }
```

Comment:
They are overridden in DataFrameColumn<T> at the moment?. Do you mean they should occur only in DataFrameColumn<T> and not in the base class? I'm not completely convinced about that yet. At the moment, the DataFrame class has a DataFrameTable which has an IList<BaseDataFrameColumn> called columns i.e. the DataFrameTable does not know the real type of the columns it holds => therefore, the approach at the moment is to add APIs on BaseDataFrameColumn and override them in the derived columns. It is possible that this will change down the line as I add more features, but for the moment I think this looks reasonable. Thoughts?